secant/Features/App/AppDestination.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  AppDestination.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 01.12.2022.
+//
+
+import Foundation
+import ComposableArchitecture
+import ZcashLightClientKit
+
+/// In this file is a collection of helpers that control all state and action related operations
+/// for the `AppReducer` with a connection to the UI navigation.
+extension AppReducer {
+    struct DestinationState: Equatable {
+        enum Destination: Equatable {
+            case welcome
+            case startup
+            case onboarding
+            case sandbox
+            case home
+            case phraseValidation
+            case phraseDisplay
+        }
+        
+        var prevDestination: Destination?
```

Comment:
I noticed that `prevDestination` and `internalDestination` seem to be modified in relation to one another. Should they be grouped in a struct that holds them together and coherent? 

Are there combinations of both of these values that are actually invalid ?

Reply:
After your comment I think we at least missed private setter.  Otherwise, Is grouping them together providing us any advantage? Both are helpers for the computed property `destination` which should be used in the first place.

```swift
        private(set) var internalDestination: Destination = .welcome
        private(set) var previousDestination: Destination?
        
        var destination: Destination {
            get { internalDestination }
            set {
                previousDestination = internalDestination
                internalDestination = newValue
            }
        }
```

Reply:
I think the grouping responds to the question: "Are there combinations of both of these values that are actually invalid?". If the answer is yes, then grouping them and ensuring state consistency is the safest to protect us from our future selves.

Reply:
Technically both variables are independent. There's always a current destination UI is on and the one that was preceding it. So answer to your question is `no`, there can't be any combination of variable values considered invalid. So wrapping it into struct wouldn't do anything or even use of enum wouldn't help here. Important enhancement was to ensure that no one can rewrite the values of both except computed property.

Reply:
Excellent